code/game/objects/items.dm

```diff
@@ -781,8 +781,12 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 		var/mob/living/carbon/human/H = user
 		skill_modifier = H.mind.get_skill_modifier(/datum/skill/mining, SKILL_SPEED_MODIFIER)
 
+		if(H.mind.get_skill_level(/datum/skill/mining) >= SKILL_LEVEL_JOURNEYMAN && prob(H.mind.get_skill_modifier(/datum/skill/mining, SKILL_PROBS_MODIFIER))) // we check if the skill level is greater than Journeyman and then we check for the probality for that specific level.

```

Comment:
this allows you to spam click a rock to do a pulse scan

Reply:
since the range and time frame to do that is so small it wouldnt be worth it

Reply:
dont alow,
see on delay.
you cant start mine if already mine this rock.

Here problem in this, if you stop mine you need wait "mining time" to start mine this rock again